### DISCUSSION TITLE
GOCD  Pipelines creation error

### DISCUSSION THREAD

**Discussion by vikramdevopsb79** [learndevopsonline/learndevopsonline#1909] [Q&A] [answered]
![Screenshot (430)](https://github.com/learndevopsonline/learndevopsonline/assets/169573156/bf356c20-3925-4b1e-97cb-338093a488c2)
![Screenshot (429)](https://github.com/learndevopsonline/learndevopsonline/assets/169573156/2757824f-1ed0-46e7-854a-5a041422f8fb)

i have done FR requirement for backend pipeline created properly , so  when i try to do  for frontend, mysql but both pipelines are not created.
but in backend pipeline is success but getting errors for all 3 files  in config repo page

**Comment by chaithanya1812** [comment, marked as answer]
Hi @vikramdevopsb79
## Issue:- Present's in PROD stage, cross check once..
## Here is your correct version to your code. It's working.
## code:- https://github.com/ActionsRoboshop/GoCD-templates/blob/main/AnsibleDeployments/check-pipeline.yml
![image](https://github.com/learndevopsonline/learndevopsonline/assets/111736742/3e52e5f6-c97f-4821-8db3-ffcc6520f521)

**Reply by vikramdevopsb79** [reply]
@chaithanya1812 
one doubt what the use of creating templates folder in code , is it used for DRY Code only  ...?   because we are already created template Pipelines from UI right .......

**Comment by chaithanya1812** [comment]
# Templates:- we used to keep dry.
# But what happens sir uses code to create templates automatically, but it doesn't work. so that sir decided to create templates manually from UI & start using them.
1- current code of your's if fine to work with project(with out templates) 
2- you can go with templates concept.

**Reply by vikramdevopsb79** [reply]
@chaithanya1812  one more doubt related in gocd installation when application code we are adding new normal user and giving permissions and switching to  that user to extract content but in expense project backend  we just created user  for application code  why ..?

**Reply by chaithanya1812** [reply]
I know why we are changing permission for gocd-agent. but why ? not expense backend.
I will grab the infromation & we will discuss in lab session, will you join?

**Reply by vikramdevopsb79** [reply]
> I know why we are changing permission for gocd-agent. but why ? not expense backend. I will grab the infromation & we will discuss in lab session, will you join?

sure on 1 pm doubt session is available to discuss ...?

**Reply by chaithanya1812** [reply]
yes
